File: clients/rust/src/models/msg_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub headers: Option<std::collections::HashMap<String, String>>,

    /// Optional partition key. Messages with the same key are routed to the same partition.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub key: Option<String>,

    pub value: Vec<u8>,
}

impl MsgIn {
    pub fn new(value: Vec<u8>) -> Self {
        Self {
            headers: None,
            key: None,
            value,
        }
    }
}

File: clients/javascript/src/models/msgIn.ts
// this file is @generated





export interface MsgIn {
    headers?: { [key: string]: string };
/** Optional partition key. Messages with the same key are routed to the same partition. */
    key?: string | null;
value: number[];
}

export const MsgInSerializer = {
    _fromJsonObject(object: any): MsgIn {
        return {
            headers: object['headers'],
            key: object['key'],
            value: object['value'],
            };
    },

    _toJsonObject(self: MsgIn): any {
        return {
            'headers': self.headers,
            'key': self.key,
            'value': self.value,
            };
    }
}